### PR TITLE
Feature: Add more options to the ingress defintion 

### PR DESCRIPTION
**Pull request by philstel**
Mostly, so it can be used with nginx-ingress and cert-manager.

The cookie affinity settings for nginx have not been tested in production though.

Let me know what you think.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -169,13 +171,15 @@ The sync relies on rclone and uses its methods of authentication to s3 per
 
 ## Kubernetes Hosting Tips
 
-The included helm chart does not have any ingress configured. The default services are nodeports you can point a load balancer to, or alter the chart to suit. For example, on AWS you might use the AWS ALB controller and configure an ingress like this:
+The helm chart does have an ingress configuration included.
```
Comment (rubenfiszel) [inline README.md:174]: good catch

Comment (philstel) [reply README.md:174]: We should maybe add a toggle for the ingress? "ingress.enabled"?

Comment (rubenfiszel) [reply README.md:174]: :+1:

Comment (philstel) [reply README.md]: I added the "ingress.enabled" option.

File: README.md [hunk 2]
```diff
@@ -102,7 +106,8 @@ enterprise:
 | windmill.app.resources | object | `{}` | Resource limits and requests for the pods |
 | windmill.app.tolerations | list | `[]` | Tolerations to apply to the pods |
 | windmill.appReplicas | int | `2` | replica for the application app |
-| windmill.baseUrl | string | `"http://localhost"` | domain as shown in browser, change to https etc based on your endpoint/ingress configuration, sets BASE_URL environment variable in app and worker container |
+| windmill.baseDomain | string | `"localhost"` | domain as shown in browser, this variable and `baseProtocol` are used as part of the BASE_URL environment variable in app and worker container and in the ingress resource, if enabled |
```
Comment (philstel) [inline README.md:109]: `baseUrl` is now split into two parts, protocol and domain. Sadly, this is a breaking change.
I hope that's fine with you, @rubenfiszel .

A more "sophisticated" reason to keep the ingress host separated from `BASE_URL` would be to have multiple domains (e.g. internal and external DNS) pointed to the same windmill instance/svc. But I am not sure how/if windmill would handle this scenario currently and what the value of `BASE_URL` would be set to in this case.

**Comment by rubenfiszel** [comment]
This require a version bump to 1.4.3 of the chart

**Comment by rubenfiszel** [comment]
And it's not really a change of the PR itself but we should probably update the aws example for the session affinity:
The setting that we use in production and that work well is:
'''
alb.ingress.kubernetes.io/target-group-attributes: stickiness.enabled=true,stickiness.lb_cookie.duration_seconds=604800,stickiness.type=lb_cookie
'''

File: README.md (hunk 2 above)
Comment (rubenfiszel) [reply README.md:108]: I think that's very reasonable! But right now you cannot have multiple domains supported easily so I think this is a good v1.

**Comment by philstel** [comment]
Okay, let me find out what to do for the version bump.

**Comment by rubenfiszel** [comment]
You just need to update Chart.yml

**Comment by rubenfiszel** [comment]
Thanks a lot!

**Pull request merged**